src/test/java/seedu/taassist/logic/parser/GradeCommandParserTest.java

```diff
@@ -0,0 +1,60 @@
+package seedu.taassist.logic.parser;
+
+import static seedu.taassist.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.taassist.logic.commands.CommandTestUtil.GRADE_VALUE_DESC_12345;
+import static seedu.taassist.logic.commands.CommandTestUtil.INVALID_GRADE_VALUE_DESC;
+import static seedu.taassist.logic.commands.CommandTestUtil.INVALID_INDEX;
+import static seedu.taassist.logic.commands.CommandTestUtil.PREAMBLE_WHITESPACE;
+import static seedu.taassist.logic.commands.CommandTestUtil.SESSION_DESC_LAB1;
+import static seedu.taassist.logic.commands.CommandTestUtil.VALID_GRADE_VALUE_12345;
+import static seedu.taassist.logic.commands.CommandTestUtil.VALID_SESSION_LAB1;
+import static seedu.taassist.logic.parser.CliSyntax.PREFIX_SESSION;
+import static seedu.taassist.logic.parser.CommandParserTestUtil.assertParseFailure;
+import static seedu.taassist.logic.parser.CommandParserTestUtil.assertParseSuccess;
+import static seedu.taassist.logic.parser.ParserUtil.MESSAGE_INVALID_GRADE;
+import static seedu.taassist.logic.parser.ParserUtil.MESSAGE_INVALID_INDEX;
+import static seedu.taassist.testutil.TestUtil.joinWithSpace;
+import static seedu.taassist.testutil.TypicalIndexes.INDEX_FIRST_STUDENT;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.taassist.logic.commands.GradeCommand;
+import seedu.taassist.model.session.Session;
+import seedu.taassist.testutil.SessionBuilder;
+
+public class GradeCommandParserTest {
+
+    private GradeCommandParser parser = new GradeCommandParser();
+
+    @Test
+    public void parse_emptyUserInput_failure() {
+        assertParseFailure(parser, PREAMBLE_WHITESPACE, String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                GradeCommand.MESSAGE_USAGE));
+    }
+
+    @Test
+    public void parse_invalidIndex_failure() {
+        String userInput = INVALID_INDEX + SESSION_DESC_LAB1 + GRADE_VALUE_DESC_12345;
+        assertParseFailure(parser, userInput, MESSAGE_INVALID_INDEX);
+    }
+
+    @Test
+    public void parse_invalidGrade_failure() {
+        String userInput = INDEX_FIRST_STUDENT + SESSION_DESC_LAB1 + INVALID_GRADE_VALUE_DESC;
+        assertParseFailure(parser, userInput, MESSAGE_INVALID_GRADE);
+    }
+
+    @Test
+    public void parse_emptySession_failure() {
+        String userInput = joinWithSpace(INDEX_FIRST_STUDENT.toString(), PREFIX_SESSION + PREAMBLE_WHITESPACE,
+                GRADE_VALUE_DESC_12345);
+        assertParseFailure(parser, userInput, Session.MESSAGE_CONSTRAINTS);
+    }
+
```

Comment:
Should there be a test with invalid session name? The session names are verified with regex `[^\\s].*`.

Reply:
Im not very good at reading regex. Is `[^\\s].*`  "anything that isnt only whitespace"? If so, this should be covered by `parse_emptySession_failure()`

Reply:
Yes :thinking: I confused it with another checker actually.